.github/workflows/sourcepkg.yml

```diff
@@ -0,0 +1,25 @@
+name: Source Package
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: "1.21"
```

Comment:
maybe use `go-version-file: "go.mod"` so we don't have to bother keeping these in sync?  (also, our go.mod says 1.20, so maybe we should update that?)